### ISSUE TITLE
.FindByAsync()

### ISSUE THREAD

**Issue by VagyokC4** [redis/redis-om-dotnet#174]
Similar to .FindByIdsAsync() except you can add the property to search.
```csharp
[Document(StorageType = StorageType.Json)]
public class Customer
{
   [Indexed] public string FirstName { get; set; }
   [Indexed] public string LastName { get; set; }
   public string Email { get; set; }
   [Indexed(Sortable = true)] public int Age { get; set; }
   [Indexed] public string[] NickNames {get; set;}
}

var ages = [12,24,36];
var entites = RedisCollection<Customer>().FindByAsync(x => x.Age, values);
```

This would probably be good coverage for the [.Contains()](https://github.com/redis/redis-om-dotnet/issues/91#issuecomment-1207505381) request attempting to do the same.

**Comment by VagyokC4** [comment]
Looks like this is being resolved LIVE issue #91.  I think we can close this out, as the .Contains() is the more natural pattern.

**Comment by slorello89** [comment]
Yeah, I think this is a bit awkward, granted it would make the expression tree easier to parse lol. As you said I started this today in #91  so I think let's just stick to that.
